### PR TITLE
add support for FreeBSD

### PR DESCRIPTION
**Pull request by murashin**
Added support for FreeBSD on setup.py.

### CODE REVIEW

**Comment by AndrewAnnex** [comment]
hey @ murashin, thanks for the contribution but it looks like a change in this contribution is breaking the rest of the builds, but I haven't really looked into why yet.
I can also see that the check for Mac/Linux/FreeBSD is fairly repetitive, so maybe it would be worth simplifying the code a bit by either making a helper function to do the check ie:
```python
def is_nix():
    return host_OS in ("Linux", "Darwin", "FreeBSD")
```
or you could also just set a variable at the top of the file:
```python
is_nix = host_OS in ("Linux", "Darwin", "FreeBSD")
```

Thanks!

File: setup.py [hunk 1]
```diff
@@ -145,12 +145,13 @@ def build_library():
...
 def move_to_root_directory():
-    sharedLib = 'spice.so' if host_OS == "Linux" or host_OS == "Darwin" else 'cspice.dll'
+    sharedLib = 'spice.so' if host_OS in ("Linux", "Darwin", "FreeBSD") else 'cspice.dll'
     destination = os.path.join(root_dir, 'spiceypy', 'utils', sharedLib)
     if not os.path.isfile(destination):
         target = os.path.join(cspice_dir, 'lib', sharedLib) \
-                if host_OS == "Linux" or host_OS == "Darwin" else \
+                if host_OS in ("Linux", "Darwin", "FreeBSD") else \
                 os.path.join(cspice_dir, 'src', 'cspice', sharedLib)
+        os.makedirs(destination)
```
Comment (murashin) [inline setup.py]: This line seems wrong...

**Comment by murashin** [comment]
Thank you for your reply, I think the line 154 caused build failures.
I agree with your idea on creating a helper function or a variable to check Unix like OS or not.
Thanks!

**Comment by AndrewAnnex** [comment]
I think you have this fixed, the red x came from a failed travis macOS build, but they have been having stability issues recently with the macOs builds so it is not a issue with the code. I restarted the build and once it passes I will merge. 

Just as a curiosity which version(s) of FreeBSD are you using? If you can confirm it is working on those versions I would like to add this info to the main readme.

thanks again!

**Comment by murashin** [comment]
I'm using FreeBSD-12 that is CURRENT (development) branch, so not released yet.
SpiceyPy works fine for several months!

Thank you!

**Comment by AndrewAnnex** [comment]
thanks for the contribution!

**Pull request merged**